core/state/statedb.go

```diff
@@ -835,6 +860,9 @@ func (s *StateDB) RevertToSnapshot(revid int) {
 	revision := s.validRevisions[idx]
 	snapshot := revision.journalIndex
 	s.arbExtraData.unexpectedBalanceDelta = new(big.Int).Set(revision.unexpectedBalanceDelta)
+	if s.arbTxFilter == txFiltered {
+		s.arbTxFilter = notFiltered
+	}
```

Comment:
there seems to be an assumption here that revert will always follow the last tx if it was filtered. This might be true, but should at least be checked - at least, check that the tx revert happened after that snapshot and/or fail taking a snapshot if we already have a reverted transaction.. something like that

Reply:
thanks for the suggestion, it does make sense to have an additional check in place to make sure revert snapshot is called in case of errors

Reply:
I've addressed this by adding a panic

Reply:
Modified the code in block_processor to include the above check. https://github.com/OffchainLabs/nitro/pull/2807/commits/b2618c992ae8a76500c7789c00efcea106c743be#diff-6e42525ac0041b38964449b89dc7af9261e780bb800266ec13b63e5c681004e4R300-R315